### ISSUE TITLE
Moving between pages, rebinding the same image fails.

### ISSUE THREAD

**Issue by ivorhoulker** [jamietre/ImageMapster#304]
Everything works fine the first time I set up mapster on my source image.  However, when I navigate to another part of the site, and then back to the mapster image, it doesn't work as expected - tooltips and hover are not present, and there errors in the console:
```
Uncaught TypeError: Cannot read property '0' of null
    at u.MapData.getAllDataForArea (mapster.js:15)
    at HTMLAreaElement.r (mapster.js:15)
    at HTMLAreaElement.n.mouseover (mapster.js:15)
```
I tried unbinding and rebinding when I navigate away and back, which just stops it working altogether instead of acting weird.
Its behaviour without unbinding is that it shifts the position of all areas down and to the right, doubles all tooltips (adding extras at the bottom of the <body>), and the mapster_wrap_n increments.
Any ideas on how I'm meant to go about doing this properly, in a site where there can be a different mapster image in each page, and you can move between them? I'd prefer not to have to keep them all mapped and hide/unhide, but is that the expected use? Thanks.

**Comment by exp-amal** [comment]
@ivorhoulker Have you got any solution for this issue?

**Comment by ivorhoulker** [comment]
Thanks for asking.  What I've done for now is to remove all the <map> elements (before, I was leaving them all present in the DOM and just loading them together at the start). Now I only load the <map> that relates to the <img> when that image is going to be loaded, and remove it when that image is removed.  This means everything works as expected, but it's a bit messier for me.
I guess that after it's used once the <map> gets excluded from being loaded again by imagemapster, and while this makes a kind of sense, I wish I knew about this behaviour so I would be able to change my approach.  Or perhaps there's a way to change this behaviour that I'm not aware of.

**Comment by exp-amal** [comment]
@ivorhoulker Can you share your code?

**Comment by exp-amal** [comment]
Solution for this issue is to load the <map> only after loading <img>.

**Comment by ivorhoulker** [comment]
In my case I'm using it in combination with twine/sugarcube2 which is an engine for writing interactive fiction.  So it provides some events that are triggered on transitions between 'passages' - basically like views in a single page application. The first one triggered is :passagestart, then before it's displayed :passagerender is triggered with the content that's about to be displayed.
```
$(document).on(':passagestart', function (ev) {
// logic for which map to load according to destination
    $('#map-areas').html($('<map name="mapname" id="mapid">... //#map-areas is outside the loaded content
```
So the content of #map-areas which contains all the map information has been replaced at this point.  So when mapster is called on the next trigger, it will find it as though it's new.  
```
$(document).on(':passagerender', function (ev) {
    $(ev.content).find('img[usemap]').each(function () {

        let altimage = $(this).data('alt-image');
        $(this).mapster(
            {
                altImage: altimage,
                altImageFill: true,
                altImageOpacity: 1,
                //etc.
            });
    });
});
```
Again, I don't think I'm doing this in the right way but it works.  If you're using ngIf I guess you're also adding and removing stuff from the DOM conditionally to get it to work, so probably the same solution?

**Comment by exp-amal** [comment]
Solution for this issue is to load 'map' elements only after loading 'img' element

**Comment by techfg** [comment]
This is likely related to #313.  Working on a PR that includes several other items but the commit that should address this [here](https://github.com/techfg/ImageMapster/commit/9b25f7afd6bd9b866097475cd7649ac2d6b21886).

**Comment by techfg** [comment]
This should be resolved as of [v1.3.0](https://github.com/jamietre/ImageMapster/releases/tag/v1.3.0).
